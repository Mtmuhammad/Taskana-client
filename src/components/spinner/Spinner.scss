@use "../../Styles/abstracts/mixins" as *;
@use "../../Styles/abstracts/variables" as *;

@keyframes spinner {
   0% {
     transform: rotate(0deg);
   }
   100% {
     transform: rotate(360deg);
   }
 }
 .loading-spinner {
   width: 150px;
   height: 150px;
   border: 10px solid #f3f3f3; /* Light grey */
   border-top: 10px solid $primary-green;
   border-radius: 50%;
   animation: spinner 1.5s linear infinite;
   margin: 30px auto;
 }