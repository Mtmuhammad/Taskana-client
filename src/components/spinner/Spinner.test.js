import React from "react";
import { render } from "@testing-library/react";
import Spinner from "./Spinner";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<Spinner />);
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<Spinner />);
    expect(asFragment()).toMatchSnapshot();
  });
});
