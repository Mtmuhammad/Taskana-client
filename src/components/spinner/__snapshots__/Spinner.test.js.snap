// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="spinner-container"
  >
    <div
      class="loading-spinner"
    />
  </div>
</DocumentFragment>
`;
